.github/workflows/macos-ci.yml

```diff
@@ -0,0 +1,48 @@
+name: macos-ci
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+env:
+  rust-version: 1.62.0
+  python-version: "3.9"
+  poetry-version: 1.1.13
+
+jobs:
+  python-only:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # pin v3.0.2
+
+      - name: Install macfuse
+        run: brew install --cask macfuse
+
+      - name: Install poetry
+        run: |
+          export POETRY_HOME="$HOME/.poetry"
+          mkdir -p "$POETRY_HOME"
+          curl -sSL https://install.python-poetry.org \
+            | python3 - --version=${{ env.poetry-version }}
+          echo "$POETRY_HOME/bin" >> "$GITHUB_PATH"
+
+      - uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb  # pin v4.0.0
+        with:
+          python-version: ${{ env.python-version }}
+          cache: poetry
+
+      - name: Install python deps
+        run: poetry install -E core -E backend
+
+      - name: Generate PyQT GUI data
+        run: poetry run python misc/generate_pyqt.py
+
+      - name: Test python
+        run: >
+          poetry run pytest
+          -n auto
+          --log-level=DEBUG
```

Comment:
better to put that in a `pytest-args` env config to stay consistent with windows/linux jobs

Reply:
not needed for the moment, but will do latter